connectors/katalog/pkg/connector/connector.go

```diff
@@ -57,3 +65,97 @@ func (r *Handler) getAssetInfo(c *gin.Context) {
 
 	c.JSON(http.StatusOK, &response)
 }
+
+func (r *Handler) reportError(errorMessage string, c *gin.Context, httpCode int) {
+	r.log.Error().Msg(errorMessage)
+	c.JSON(httpCode, gin.H{"error": errorMessage})
+}
+
+// Enables writing of assets to katalog. The different flows supported are:
+// (a) When DestinationAssetID is specified:
+//     Then a destination asset id is created with name : <DestinationAssetID>
+// (b) When DestinationAssetID is not specified but ResourceMetadata.Name of source asset is specified:
+//     Then an asset is created with name: ResourceMetadata.Name-<RANDOMSTRING_LENGTH_4>
+// (c) When DestinationAssetID and ResourceMetadata.Name of source asset are not specified:
+//     Then an asset is created with name: fybrik-asset-<RANDOMSTRING_LENGTH_4>
+func (r *Handler) createAssetInfo(c *gin.Context) {
+	// Parse request
+	var request datacatalog.CreateAssetRequest
+	if err := c.ShouldBindJSON(&request); err != nil {
+		r.reportError("Error during ShouldBindJSON in createAssetInfo"+err.Error(), c, http.StatusBadRequest)
+		return
+	}
+
+	r.log.Info().Msg("CreateAssetRequest object received: " + fmt.Sprintf(" %+v", request))
+
+	if request.DestinationCatalogID == "" {
+		r.reportError("Invalid DestinationCatalogID in request", c, http.StatusBadRequest)
+		return
+	}
+
+	var assetName string
+	var namespace string
+	var err error
+	if request.DestinationAssetID != "" {
+		namespace, assetName = request.DestinationCatalogID, request.DestinationAssetID
+	} else {
+		if request.ResourceMetadata.Name != "" {
+			namespace, assetName = request.DestinationCatalogID, request.ResourceMetadata.Name
+			assetName, err = utils.GenerateUniqueAssetName(namespace, assetName, &r.log, r.client)
+		} else {
+			namespace = request.DestinationCatalogID
```

Comment:
no need to set namespace to the same value inside if statements

Reply:
Fixed. Please check. Thanks.